goas/atomic_asset_transfer/proof_statements/src/zone_funds.rs

```diff
@@ -19,3 +19,19 @@ pub struct SpendFundsPrivate {
     /// Merkle root of balances in the zone
     pub balances_root: [u8; 32],
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct MergePrivate {
+    // The note we're spending
+    pub funds_note: cl::PartialTxInputWitness,
+    // The zone note that is authorizing the spend
+    pub zone_note: cl::PartialTxOutputWitness,
+    // The event emitted by the zone that authorizes spending this note
+    pub merge_event: common::events::Merge,
+    // Path to the zone output events root
+    pub merge_event_state_path: Vec<cl::merkle::PathNode>,
```

Comment:
Do we need this merge event? If we follow the idea of tracking total zone funds in the zone state, and the zone state asserting that the total output funds exactly matches the expected output zone funds.

Then this merge constraint reduces to merely checking that the zone state was an input.